test/e2e/e2e.go

```diff
@@ -302,6 +303,24 @@ func setupSuite() {
 		nodeKiller := framework.NewNodeKiller(framework.TestContext.NodeKiller, c, framework.TestContext.Provider)
 		go nodeKiller.Run(framework.TestContext.NodeKiller.NodeKillerStopCh)
 	}
+
+	// Allow anonymous access to `/metrics` in tests
```

Comment:
hmm... is this what we expect anyone accessing scheduler metrics via the kube-apiserver proxy to do? this also grants access to apiserver /metrics to all anonymous users